book/machinelearning.qmd

```diff
@@ -4,114 +4,215 @@ abstract: TODO (150-200 WORDS)
+
+Cross-validation can also be stratified, which ensures that a variable of interest will have the same distribution in each fold as in the original data.
+This is important, and often recommended, in survival analysis to ensure that the proportion of censoring in each fold is representative of the full dataset.
+<!-- FIXME: CITE OUR SURVIVAL BENCHMARK PAPER -->
+
+Repeating resampling experiments with multiple models is referred to as a *benchmark experiment*.
+A benchmark experiment compares models by evaluating their performance on *identical* data, which means the same resampling strategy and folds should be used for all models.
+Determining if one model is actually better than another is a surprisingly complex topic [@Demsar2006; @Dietterich1998; @Nadeau2003] and is out of scope for this book, instead any benchmark experiments performed in this book are purely for illustrative reasons and no results are expected to generalize outside of these experiments.
+
+## Optimization {#sec-ml-opt}
+
+@sec-ml-models introduced model hyperparameters, which control how training and prediction algorithms are run.
+Setting hyperparameters is a critical part of model fitting and can significantly change model performance.
+*Tuning* is the process of using internal benchmark experiment to automatically select the optimal hyper-parameter configuration.
+For example, the number of trees, $m_r$ in a random forest (@sec-surv-ml-models-ranfor) is a potential hyperparameter to tune.
+This hyperparameter may be tuned over a range of values, say $[10, 1000]$ or over a discrete subset, say $\{10, 100, 1000\}$, for now assume the latter.
+Three random forests with $10$, $100$, and $1000$ trees respectively are compared in a benchmark experiment.
+The number of trees that results in the model with the optimal performance is then selected for the hyperparameter value going forward.
+*Nested resampling* is a common method to prevent overfitting that could occur from using overlapping data for tuning, training, or testing.
+Nested resampling is the process of resampling the training set again for tuning and then the optimal model is refit on the entire training data (@fig-ml-nested).
+
+![An illustration of nested resampling. The large blocks represent three-fold CV for the outer resampling for model evaluation and the small blocks represent four-fold CV for the inner resampling for HPO. The light blue blocks are the training sets and the dark blue blocks are the test sets. Image and caption from @Becker2024 (CC BY-NC-SA 4.0).](https://mlr3book.mlr-org.com/chapters/chapter4/Figures/mlr3book_figures-11.svg){#fig-ml-nested}
+
+## Conclusion
+
+:::: {.callout-warning icon=false}
+
+## Key takeaways
+
+* Machine learning tasks define the predictive problem of interest;
+* Regression tasks make continuous predictions and classification tasks make discrete predictions;
```

Comment:
not true for classif task that predicts probability, its bounded but not discrete